### PR TITLE
feat: adding __repr__  method to CompactionMetrics

### PR DESCRIPTION
**Pull request by avr2002**
Issue Link: https://github.com/lancedb/lance/issues/1373

### CODE REVIEW

File: python/python/lance/lance/__init__.pyi [hunk 1]
```diff
@@ -32,6 +32,14 @@ class CompactionMetrics:
...
     files_removed: int
     files_added: int
+    
+    def __repr__(self):
+        return f"""
+        Fragments Removed: {self.fragments_removed}
+        Fragments Added: {self.fragments_added}
+        Files Removed: {self.files_removed}
+        Files Added: {self.files_added}
+    """
```
Comment (wjones127) [inline python/python/lance/lance/__init__.pyi:42]: I don't think function implementations are meant to be put into `.pyi` files, which are type stubs. 

`CompactionMetrics` is implemented in Rust here: 

https://github.com/lancedb/lance/blob/1b8adb56f209eb0690c0a2e95ea06896188e656a/python/src/dataset/optimize.rs#L78

Could you implement the method there in Rust? Here is an example Rust implementation of a `__repr__` method:

https://github.com/lancedb/lance/blob/1b8adb56f209eb0690c0a2e95ea06896188e656a/python/src/dataset/optimize.rs#L113-L119

**Review by wjones127 — CHANGES_REQUESTED** [review]
Thanks for making a PR! In order to make this ready to merge, could you make sure to add a unit test to validate the repr method works?

File: python/python/lance/lance/__init__.pyi (hunk 1 above)
Comment (avr2002) [reply python/python/lance/lance/__init__.pyi:42]: Thanks for getting back, I really appreciate the support. This is my first time contributing to open-source project. It'll try to implement this and get back.

**Comment by wjones127** [comment]
Hi @avr2002, I'm cleaning up our PRs and it's been a while since any progress is made here. If you end up getting back to this, feel free to re-open the PR. Your contribution would be most welcome.

**Pull request closed without merging**